### ISSUE TITLE
🚀 Feature:  Option to selectively include related fields in relational database queries

### ISSUE THREAD

**Issue by aliazimoshan** [appwrite/appwrite#5494]
### 🔖 Feature description

It would be helpful to have an option to selectively include related fields in relational database queries. For example, we could specify which related fields to include in the query or whether to include them at all (like populate in mongoDB). This would make it easier to optimize queries and reduce data usage.

### 🎤 Pitch

Currently, when a relational field is added to a document in the Appwrite database, all related fields are returned when the document is queried. This behavior can be problematic when we only need certain fields and returning all related fields can result in unnecessary data usage and processing time.

I believe that adding this option would greatly improve the flexibility and efficiency of the Appwrite database, and would be a valuable addition to the platform.

Thank you for your attention to this matter.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by Vedsaga** [comment]
@aliazimoshan so `select` query already exist...  https://appwrite.io/docs/queries

**Comment by omzi** [comment]
@joeyouss Let me further explain @aliazimoshan's issue (I understand it because I was looking for a method to achieve that). Let's say I have a Posts collection with the following schema.

```js
title: String
content: String
isDraft: Boolean
authorId: String<Ref:User>
```

The `authorId` is a reference to a record on the Users collection. When fetching a document from the Posts collection, you can select the fields you want to be returned using `Query.select([])`. If you want to populate the `authorId` field with data from the Users collection, the only option you have currently is to make a second request which is not performant.

I suggest a new Query method be made to be able to populate fields. Maybe something like `Query.populate(fieldToPopulate1, fieldToPopulate2, ...)`? My current workaround is using `database.getDocument` when I need to populate ONE document and `databases.listDocuments` when I needed to populate multiple documents.

@christyjacob4 @stnguyen90 Please could you guys look into this issue? Thanks :)

**Comment by stnguyen90** [comment]
@omzi, it sounds like you're describing [relationships](https://appwrite.io/docs/databases-relationships) which was added in Appwrite version 1.3.

**Comment by aliazimoshan** [comment]
Hi everyone,

Thank you for your valuable insights and suggestions regarding this feature request.

I appreciate the detailed explanation provided by the community member who raised this concern. In the new version of Appwrite's relational database, there is an issue with the default behavior of returning all related data when querying a collection. This behavior can impact performance, particularly in scenarios involving large datasets.

As an example, let's consider a "Post" collection with fields like "title," "description," "image," "creatorId," and "likes." The "creatorId" and "likes" fields are relational, referencing the "User" collection. Currently, when retrieving a post, the response includes the complete user data for the creator and all the users who liked the post.

While the select query feature allows us to remove specific fields from the response, it doesn't provide an ideal solution in cases where we need additional information. For instance, it is essential to retrieve the number of likes and, ideally, the corresponding user IDs who liked a post.

To address this, I propose considering a mechanism similar to MongoDB's population feature. This would allow us to selectively populate relational fields based on our requirements. By providing control over which fields are populated, we can optimize query performance while still obtaining necessary information such as the IDs of users who liked a post.

I kindly request the expertise of @joeyouss and @stnguyen90 in reviewing this proposal. The addition of a selective population feature for relational fields would greatly enhance the flexibility, usability, and performance of the Appwrite database.

Thank you all for your contributions and collaboration on this matter.

Best regards,

**Comment by abnegate** [comment]
Closing as `select` queries are available from 1.3.x onwards
